module/purpose/FieldPlayer/src/FieldPlayer.cpp

```diff
@@ -90,27 +92,50 @@ namespace module::purpose {
                     return;
                 }
 
+                // Make an ignore list with the goalie, if they exist
+                std::vector<unsigned int> ignore_ids{};
+                // Add inactive robots to the ignore list
+                for (const auto& robot : (robots ? robots->robots : std::vector<message::localisation::Robot>{})) {
+                    if (!robot.purpose.active) {
+                        ignore_ids.push_back(robot.purpose.player_id);
+                    }
+                }
+
+                // Find who has the ball, if any
+                // If there are no robots, use an empty vector
+                Who ball_pos = utility::strategy::ball_possession(ball->rBWw,
+                                                                  (robots ? *robots : Robots{}),
+                                                                  field.Hfw,
+                                                                  sensors.Hrw,
+                                                                  cfg.ball_threshold,
+                                                                  cfg.equidistant_threshold,
+                                                                  global_config.player_id,
+                                                                  ignore_ids);
+
                 // If we have robots, determine if we are closest to the ball
                 // Otherwise assume we are alone and closest by default
+                // Only consider the goalie if the ball is in the defending third
+                Eigen::Vector3d rBFf     = field.Hfw * ball->rBWw;
+                double defending_third_x = fd.dimensions.field_length / 3.0;
+                if (robots && rBFf.x() < defending_third_x) {
+                    for (const auto& robot : (robots ? robots->robots : std::vector<message::localisation::Robot>{})) {
```

Comment:
```suggestion
                    for (const auto& robot : robots->robots) {
```
The `if` above already checks if `robots` is truthy